include/SDL3/SDL_ioasync.h

```diff
@@ -0,0 +1,500 @@
+/*
+  Simple DirectMedia Layer
+  Copyright (C) 1997-2024 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+
+/* WIKI CATEGORY: IOAsync */
+
+/**
+ * # CategoryIOAsync
+ *
+ * SDL offers a way to perform i/o _asynchronously_. This allows an app to
+ * read or write files without waiting for data to actually transfer; the
+ * functions that request i/o never block while the request is fulfilled.
+ *
+ * Instead, the data moves in the background and the app can check for
+ * results at their leisure.
+ *
+ * This is more complicated that just reading and writing files in a
+ * synchronous way, but it can allow for more efficiency, and never having
+ * framerate drops as the hard drive catches up, etc.
+ *
+ * The general usage pattern for async i/o is:
+ *
+ * - Create one or more SDL_IOAsyncGroup.
```

Comment:
I still think we should have some sort of interface for this into SDL_Storage, even if it returns SDL_Unsupported() for now.

Reply:
What I've learned from this is that these sort of APIs are not going to be implemented by anything in a serious way, so beyond the built-in filesystem Storage being backed by this PR's code, I don't see anyone else implementing this.

Maybe that's enough, though.

Reply:
Yeah, getting an async interface for title storage is probably important.

Reply:
Most cloud storage APIs seem to provide their own async operations out of the box.